### PR TITLE
[Rollup] Replace RollupIT with a ESRestTestCase version

### LINKED ISSUE #30232: [CI] RollupIT#testBig fails
I can reproduce this locally.

```
./gradlew :x-pack:plugin:rollup:internalClusterTest \
-Dtests.seed=38E445F745E9BA6E \
-Dtests.class=org.elasticsearch.xpack.rollup.RollupIT \
-Dtests.method="testBig" \
-Dtests.security.manager=true \
-Dtests.jvms=8 \
-Dtests.locale=luy \
-Dtests.timezone=Pacific/Apia
```

Assertion error

```
FAILURE 7.99s | RollupIT.testBig <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <89L>
   >      but: was <88L>
   >    at __randomizedtesting.SeedInfo.seed([38E445F745E9BA6E:37DA9EAD70777305]:0)
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   >    at org.elasticsearch.xpack.rollup.RollupIT.testBig(RollupIT.java:429)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:564)
   >    at java.base/java.lang.Thread.run(Thread.java:844)
```

### LINKED ISSUE #31258: RollupIT disables all thread leak detection
Currently the main rollup integration test disables all thread leak detection: https://github.com/elastic/elasticsearch/blob/2efd22454a62e7387c51eb0cd973e5ff7a3fb09f/x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/RollupIT.java#L64-L65

I think that this should be unnecessary, I think that the only thread that should be filtered out here is the `ObjectCleanerThread` from Netty (relates elastic/elasticsearch#31232). It looks like this was added for the `trigger_engine_scheduler` thread. Can we look into this and see if the thread leak none can be removed and reduced to filter the `ObjectCleanerThread` only (extend `XPackIntegTestCase`)?

### PR DESCRIPTION
**Pull request by polyfractal**
The old RollupIT was a node IT, and flaky for a number of reasons. This new version is an ESRestTestCase and should be a little more robust.  It shoould also be easier to add more varied tests in the future with this setup.

This was added to the multi-node QA tests as that seemed like the most appropriate location.  It didn't seem necessary to create a whole new QA module.

Note: The only test that was ported was the "Big" test for validating a larger dataset.  The rest of the tests are represented in existing yaml tests.

Closes #31258
Closes #30232
Related to #30290

### CODE REVIEW

File: x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+    private void assertRollUpJob(final String rollupJob) throws Exception {
+        final Matcher<?> expectedStates = anyOf(equalTo("indexing"), equalTo("started"));
+        waitForRollUpJob(rollupJob, expectedStates);
+
+        // check that the rollup job is started using the RollUp API
+        final Request getRollupJobRequest = new Request("GET", "_xpack/rollup/job/" + rollupJob);
+        Map<String, Object> getRollupJobResponse = toMap(client().performRequest(getRollupJobRequest));
+        Map<String, Object> job = getJob(getRollupJobResponse, rollupJob);
+        if (job != null) {
+            assertThat(ObjectPath.eval("status.job_state", job), expectedStates);
```
Comment (jimczi) [inline x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java]: this fails the test compilation, should it be:
` assertThat(ObjectPath.eval("status.job_state", job), equalTo(expectedStates));`?

File: x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+        // check that the rollup job is started using the Tasks API
+        final Request taskRequest = new Request("GET", "_tasks");
+        taskRequest.addParameter("detailed", "true");
+        taskRequest.addParameter("actions", "xpack/rollup/*");
+        Map<String, Object> taskResponse = toMap(client().performRequest(taskRequest));
+        Map<String, Object> taskResponseNodes = (Map<String, Object>) taskResponse.get("nodes");
+        Map<String, Object> taskResponseNode = (Map<String, Object>) taskResponseNodes.values().iterator().next();
+        Map<String, Object> taskResponseTasks = (Map<String, Object>) taskResponseNode.get("tasks");
+        Map<String, Object> taskResponseStatus = (Map<String, Object>) taskResponseTasks.values().iterator().next();
+        assertThat(ObjectPath.eval("status.job_state", taskResponseStatus), expectedStates);
```
Comment (jimczi) [inline x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java]: Same here

File: x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+        Map<String, Object> clusterStateResponse = toMap(client().performRequest(clusterStateRequest));
+        List<Map<String, Object>> rollupJobTasks = ObjectPath.eval("metadata.persistent_tasks.tasks", clusterStateResponse);
+
+        boolean hasRollupTask = false;
+        for (Map<String, Object> task : rollupJobTasks) {
+            if (ObjectPath.eval("id", task).equals(rollupJob)) {
+                hasRollupTask = true;
+
+                final String jobStateField = "task.xpack/rollup/job.state.job_state";
+                assertThat("Expected field [" + jobStateField + "] to be started or indexing in " + task.get("id"),
```
Comment (jimczi) [inline x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java]: and here ;)

File: x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+    private void waitForRollUpJob(final String rollupJob, final Matcher<?> expectedStates) throws Exception {
+        assertBusy(() -> {
+            final Request getRollupJobRequest = new Request("GET", "_xpack/rollup/job/" + rollupJob);
+            Response getRollupJobResponse = client().performRequest(getRollupJobRequest);
+            assertThat(getRollupJobResponse.getStatusLine().getStatusCode(), equalTo(RestStatus.OK.getStatus()));
+
+            Map<String, Object> job = getJob(getRollupJobResponse, rollupJob);
+            if (job != null) {
+                assertThat(ObjectPath.eval("status.job_state", job), expectedStates);
```
Comment (jimczi) [inline x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java]: here as well ?

File: x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+        bulkRequest.addParameter("refresh", "true");
+        bulkRequest.setJsonEntity(bulk.toString());
+        client().performRequest(bulkRequest);
+        // create the rollup job
+        final Request createRollupJobRequest = new Request("PUT", "/_xpack/rollup/job/rollup-job-test");
+        createRollupJobRequest.setJsonEntity("{"
+            + "\"index_pattern\":\"rollup-*\","
+            + "\"rollup_index\":\"results-rollup\","
+            + "\"cron\":\"*/1 * * * * ?\","             // fast cron and big page size so test runs quickly
+            + "\"page_size\":200,"
```
Comment (jimczi) [inline x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java]: We should test multiple pages, how slow is it if you set the `page_size` to 20 for instance ?

**Review by jimczi — COMMENTED** [review]
Thanks @polyfractal ! I left some comments

File: x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java (hunk 5 above)
Comment (polyfractal) [reply x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java]: ++ I dont think multiple pages will be bad.  The major slowdown is waiting for the cron to tick over (e.g. if it's 30s and you get unlucky you may be waiting the full time).

File: x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java (hunk 1 above)
Comment (polyfractal) [reply x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java]: It fails compilation for you?  Looks ok on my side... I pilfered this from `FullClusterRestartIT` tests that Tanguy added recently. `expectedStates` is defined above a few lines:

`final Matcher<?> expectedStates = anyOf(equalTo("indexing"), equalTo("started"));`

File: x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java (hunk 4 above)
Comment (polyfractal) [reply x-pack/qa/multi-node/src/test/java/org/elasticsearch/multi_node/RollupIT.java]: here `expectedStates` is passed in as a `final Matcher<?> expectedStates`

**Comment by polyfractal** [comment]
Jenkins, run gradle build tests

**Review by jimczi — APPROVED** [review]
CI still fails due to a compilation error with `assertThat`. I left a comment where it happens but no need for another review, we just need a green CI.

**Comment by polyfractal** [comment]
Just pushed a new derivation using `isOneOf()` with a string array.  :crossed_fingers:

**Pull request merged**